materialize-mysql/driver.go

```diff
@@ -734,21 +769,27 @@ func (d *transactor) Store(it *m.StoreIterator) (_ m.StartCommitFunc, err error)
 			defer txn.Rollback()
 
 			for _, b := range d.bindings {
-				if !b.mustMerge {
-					// Keys that don't already exist are inserted directly into the target table.
-					continue
-				}
+				if b.mustDelete {
```

Comment:
Does this work if there are deletions and merge updates in the same transaction for the same binding?

Reply:
Good point! I initially made the same mistake on postgres as well and fixed it later...
So I just updated the deletions fixture to include a transaction during which we delete, update and insert at the same time to make sure this doesn't happen